core/src/main/java/org/hisp/dhis/android/core/sms/domain/interactor/ConfigCase.kt

```diff
@@ -0,0 +1,191 @@
+                moduleEnabled,
+                gateway,
+                waitingForResult,
+                resultSender,
+                resultWaitingTimeout
+            )
+        }
+
+    /**
+     * Set if the SDK has to wait for the result or not.
+     * @param enabled
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setWaitingForResultEnabled(enabled: Boolean): Completable {
+        return localDbRepository.setWaitingForResultEnabled(enabled)
+    }
+
+    /**
+     * Set the number to receive messages from. Messages from other senders will not be read.
+     * @param number The sender number.
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setConfirmationSenderNumber(number: String): Completable {
+        return localDbRepository.setConfirmationSenderNumber(number)
+    }
+
+    /**
+     * Set the time in seconds to wait for the result.
+     * @param timeoutSeconds Time in seconds.
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setWaitingResultTimeout(timeoutSeconds: Int): Completable {
+        return localDbRepository.setWaitingResultTimeout(timeoutSeconds)
+    }
+
+    /**
+     * Set the gateway number to send the SMS. This is a required parameter before sending SMS.
+     * @param gatewayNumber The gateway numberf
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setGatewayNumber(gatewayNumber: String?): Completable {
+        return if (gatewayNumber.isNullOrEmpty()) {
+            Completable.error(IllegalArgumentException("Gateway number can't be empty"))
+        } else localDbRepository.setGatewayNumber(gatewayNumber)
+    }
+
+    /**
+     * Delete the gateway number to send the SMS.
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun deleteGatewayNumber(): Completable {
+        return localDbRepository.deleteGatewayNumber()
+    }
+
+    /**
+     * Set if SMS Module is enabled or not. It is required to enable it before using it.
+     * @param enabled If the module is enabled or not
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setModuleEnabled(enabled: Boolean): Completable {
+        return localDbRepository.setModuleEnabled(enabled)
+    }
+
+    /**
+     * Set a new MetadataDownload configuration.
+     * @param metadataIdsConfig Configuration with the metadata ids.
+     * @return `Completable` that completes when the configuration is changed.
+     */
+    fun setMetadataDownloadConfig(metadataIdsConfig: GetMetadataIdsConfig?): Completable {
+        return if (metadataIdsConfig == null) {
+            Completable.error(IllegalArgumentException("Received null config"))
+        } else localDbRepository.setMetadataDownloadConfig(metadataIdsConfig)
+    }
+
+    val metadataDownloadConfig: Single<GetMetadataIdsConfig>
```

Comment:
By default, functions starting with `fun getX()` are translated to `val x`, which follows Java conventions but is a bit weird... What do you think? Should we stick to `fun getMetadataDownloadConfig()`?

Reply:
Agree, the `fun` approach seems more solid and clear